ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -348,6 +349,7 @@ protected void initChannel(final SocketChannel ch) {
 
               if (!isPeerAllowed(connection)) {
                 connection.disconnect(DisconnectReason.UNKNOWN);
+                peerDiscoveryAgent.dropPeer(connection.getPeer());
```

Comment:
Previously we were dropping all peers who disconnected.  Now, we're explicitly dropping a subset of disconnected peers.